test/es7/asyncawait-syntax.js

```diff
@@ -128,6 +128,16 @@ var tests = [
             var d = async (a, b) => { };
         }
     },
+    {
+        name: "[no LineTerminator here] after `async` in grammar",
+        body: function () {
+            assert.throws(function () { eval("async\nfunction af() { }"); }, ReferenceError, "AsyncFunctionDeclaration", "'async' is undefined");
```

Comment:
I agree a friendlier error message would be great here.  However, `throw` is a reserved word though so it is easy to know at parse time that `throw` syntax was incorrect. `async` is a pseudo-keyword, allowed to be a normal identifier when it is used not in async function syntax.  So parsing needs to continue you onward when it doesn't parse like an async function meaning the error is reported later so state will need to be maintained in parsing all async grammar productions and then interrogated at each possible syntax failure to know to print the correct message.

Our parsing error reporting and messages and runtime error messages could use a lot of improvement but let's not tackle this case by case.  This should be approached as a priority investment if we deem it a priority, which after completed will set the precedent for new errors to maintain a high level of quality case by case.

We can discuss this offline in Tuesday's stand up when I'm back.